imports/plugins/core/layout/client/templates/layout/wallet/wallet.js

```diff
@@ -0,0 +1,213 @@
+/* eslint no-undef: 0 */
+import { Meteor } from "meteor/meteor";
+import { Template } from "meteor/templating";
+import { Reaction } from "/client/api";
+import { Accounts, Packages, Wallet, Shops } from "/lib/collections";
+import Alert from "sweetalert2";
+import bcrypt from "bcrypt-nodejs";
+
+Template.wallet.onCreated(function bodyOnCreated() {
+  this.state = new ReactiveDict();
+  this.state.setDefault({
+    details: {
+      balance: 0,
+      transactions: []
+    }
+  });
+  this.autorun(() => {
+    this.subscribe("transactionInfo", Meteor.userId());
+    const transactionDetail = Wallet.find().fetch();
+    this.state.set("details", transactionDetail[0]);
+  });
+});
+const getPaystackSettings = () => {
+  const settings = Packages.findOne({
+    name: "paystack-paymentmethod",
+    shopId: Reaction.getShopId()
+  });
+  return settings;
+};
+
+const getExchangeRate = () => {
+  const shop = Shops.find(Reaction.getShopId()).fetch();
+  return shop[0].currencies.NGN.rate;
+};
+
+const finalizeDeposit = (paystackMethod) => {
+  paystackMethod.transactions.amount = paystackMethod.amount / (100 * getExchangeRate());
```

Comment:
You are mutating this object. Avoid this because it can lead to code smell. Use functional approach, please 😀